packages/common/src/converter/payload-converter.ts

```diff
@@ -276,34 +260,34 @@ export class ProtobufJsonPayloadConverter extends ProtobufPayloadConverter {
 
   public fromDataSync<T>(content: Payload): T {
     const { messageType, data } = this.validatePayload(content);
-    return protoJsonSerializer.fromProto3JSON(messageType as any, JSON.parse(str(data))) as any as T;
+    return protoJsonSerializer.fromProto3JSON(messageType, JSON.parse(str(data))) as unknown as T;
   }
 }
 
-function isProtobufMessageType(type: unknown): type is ProtobufCodec {
+function isProtobufType(type: unknown): type is Type {
   return (
     isRecord(type) &&
     type.constructor.name === 'Type' &&
     'parent' in type &&
-    typeof (type as unknown as ProtobufCodec).name === 'string' &&
-    typeof (type as unknown as ProtobufCodec).create === 'function' &&
-    typeof (type as unknown as ProtobufCodec).encode === 'function' &&
-    typeof (type as unknown as ProtobufCodec).decode === 'function'
+    typeof (type as unknown as Type).name === 'string' &&
+    typeof (type as unknown as Type).create === 'function' &&
+    typeof (type as unknown as Type).encode === 'function' &&
+    typeof (type as unknown as Type).decode === 'function'
```

Comment:
Can we cast only once here?

Reply:
Removed casting:

![image](https://user-images.githubusercontent.com/251288/147423308-347e78d5-60a2-4574-89f8-1d34fa712ca5.png)